src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,41 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  showModal: boolean;
+  setShowModal: (close: boolean) => void;
+  todo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  showModal,
+  setShowModal,
+  todo,
+}) => {
+  const [user, setUser] = useState<User>();
+
+  function getCurrentUser() {
+    getUser(todo.userId).then(setUser);
+  }
+
+  useEffect(() => {
+    getCurrentUser();
+  }, []);
```

Comment:
The `useEffect` hook should include `todo.userId` in its dependency array to ensure that `getCurrentUser` is called whenever the `todo.userId` changes.